apps/bbsync/query.py

```diff
@@ -219,17 +248,29 @@ def generate_summary(self):
         # 3) filter out the empty CVE IDs
         #    there can be one in case we are removing the CVE ID
         cve_ids = [cve_id for cve_id in cve_ids if cve_id]
-        # 4) filter out eventual duplicates | sort by CVE ID | stringify
-        cve_ids = " ".join(sorted(list(set(cve_ids)), key=cve_id_comparator))
-        # 5) add trailing space delimiter in case of non-empty CVE IDs
-        cve_ids = cve_ids + " " if cve_ids else cve_ids
+        # 4) filter out eventual duplicates | sort by CVE ID
+        cve_ids = sorted(list(set(cve_ids)), key=cve_id_comparator)
 
         components = (
             ": ".join([component for component in self.flaw.components]) + ": "
             if self.flaw.components
             else ""
         )
-        self._query["summary"] = embargoed + cve_ids + components + self.flaw.title
+
+        # try to compose the summary
+        # until it is short enough
+        description = self.flaw.title
+        while True:
+
+            cve_string = " ".join(cve_ids) + " " if cve_ids else ""
+            summary = f"{embargoed}{cve_string}{components}{description}"
+
+            if len(summary) <= MAX_SUMMARY_LENGTH:
+                break
+
+            cve_ids, description = summary_shorten(cve_ids, description)
+
+        self._query["summary"] = summary
```

Comment:
If we have a long list of `components`, we may reach a point when `summary_shorten` can't shorten `cve_ids` and `description` further but we are still over the limit, ending in an infinite loop.

It might be a good idea to use a variable to keep track of whether we are still reducing the length at each step, and if we reach the limit (length is not reducing anymore but we are still over the max length) just exit the loop and shorten it the barbarian way
```py
summary = summary[:MAX_SUMMARY_LENGTH-3] + "..."
```

Reply:
Makes sense

Reply:
@dmonzonis so added a condition to error out instead of shortening in case we already have the very last CVE ID and the description is minimal - so the reason of the length are the components. However, I see that this is actually no change from the original behavior which would try to shorten once more and fail with the completely same error. Do you really think that there is any change needed?

Reply:
It should be okay now, but I still think that relying on a method that is called inside the `while True` loop to raise an error in order to exit the loop can be dangerous. If someone else in the future modifies this method without knowing this, and removes the `raise`, we could end up with an infinite loop, which would be a rare case but still possible. I still think a safeguard in the loop doesn't add a lot of complexity and can prevent this.

But I'll leave it up to you, the code as it is now works well.